src/main/java/parsers/agmart/AgmartCrawlingService.java

```diff
@@ -0,0 +1,85 @@
+    public List<CrawlCommodityPriceDto> handleRequest(Long date) {
+      HashSet<String>state_ids=new HashSet<String>();
+        HashSet<String>city_ids=new HashSet<String>();
+        HashMap<String,HashSet<String>> state_city=new HashMap<String,HashSet<String>>();
+        try {
+            Document doc = Jsoup.connect("https://agmart.in/commodity-market").get();
+            AgmartParser agmartParser=new AgmartParser();
+            state_ids=agmartParser.getState_ids(doc.toString());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        for(String state_id:state_ids){
+            HttpRequestDto httpRequestDto=buildRequest_city(state_id);
+            HttpClientPool httpClientPool=new HttpClientPool();
+            AgmartParser agmartParser=new AgmartParser();
+            try {
+                HttpResponseDto responseDto=httpClientPool.executeRequest(httpRequestDto);
+                if(responseDto.getSuccessful()){
+                    state_city.put(state_id,agmartParser.getCityids(responseDto.getResponseString(),state_id));
+                }
+            } catch (IOException e) {
+                System.out.println("error");
+                e.printStackTrace();
+            }
+        }
+        List<HttpRequestDto> httpRequestDtosList = new ArrayList<>();
+        ParserCallableTaskExecutorService parserCallableTaskExecutorService =
+                new ParserCallableTaskExecutorService();
+        List<CrawlCommodityPriceDto> crawlCommodityPriceDtoList =
+                null;
+        try {
+            AgmartParser.setDate(date);
+            for(String state_id:state_city.keySet()){
+                for (String city_id: state_city.get(state_id)){
+                    httpRequestDtosList.add(buildRequest(date,state_id,city_id));
+                }
+            }
+            crawlCommodityPriceDtoList = parserCallableTaskExecutorService.executeTask(httpRequestDtosList,
+                    new AgmartParser());
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        if(crawlCommodityPriceDtoList==null){
+            return new ArrayList();
+        }
+        return crawlCommodityPriceDtoList;
+
+    }
+    public HttpRequestDto buildRequest(Long date,String state_id,String city_id){
+        String url= CommodityPriceSource.AGMART.getUrl();
+        String dateStr = DateUtils.format(new SimpleDateFormat("yyyy-MM-dd"), date);
+        String payload =String.format("state_id=%s&market_id=%s&date=%s",state_id,city_id,"2021-08-04");
+        Map<String, String> headers = HttpHeaderUtils.getApplicationFormURLEncodedHeaders();
+        return HttpRequestDto.Builder.httpRequestDto()
+                .withRequestType(RequestType.POST)
+                .withUrl(url)
+                .withHeaders(headers)
+                .withPayload(payload)
+                .build();
+    }
+    public HttpRequestDto buildRequest_city(String state_id){
```

Comment:
In java we user camelCase for method and variable name line state_id would be stateId